dsdl_compiler/libcanard_dsdl_compiler/code_type_template.tmpl

```diff
@@ -76,19 +76,19 @@ uint32_t ${type_name}_encode_internal(${type_name}* source,
                     %if f.bitlen < 8
     ${f.array_max_size_bit_len}
     //  - Add array length, last item, but bitlen < 8.
-    canardEncodeScalar(msg_buf, offset, ${f.array_max_size_bit_len}, (void*)&source->${'%s' % ((f.name + '.len'))});
+    canardEncodePrimitive(msg_buf, offset, ${f.array_max_size_bit_len}, (void*)&source->${'%s' % ((f.name + '.len'))});
     offset += ${f.array_max_size_bit_len};
                     %else
     if (! root_item)
```

Comment:
The TAO (tail array optimization) logic needs to be removed (finally!). Would you like to do it in a new PR?

Reply:
Would prefer to do it in another PR, has created #108.